### ISSUE TITLE
Canvas crashing on scene load

### ISSUE THREAD

**Issue by Zamrod** [caewok/fvtt-elevated-vision#34]
Errors when a scene loads and canvas never appears. This seems to only happen in scenes with walls. If there are no walls, the scene loads. The errors are below

```
TypeError: Cannot read properties of undefined (reading 'heightWalls')
[Detected 2 packages: elevatedvision, lib-wrapper]
    at LightSource._updateEVLightUniformsLightSource (lighting.js:460:48)
    at LightSource._updateIlluminationUniformsLightSource (lighting.js:439:8)
    at 🎁LightSource.prototype._updateIlluminationUniforms#0 (libWrapper-wrapper.js:187:52)
    at LightSource._updateUniforms (foundry.js:49178:10)
    at LightSource._initializeShaders (foundry.js:49024:10)
    at LightSource.initialize (foundry.js:48915:31)
    at AmbientLight.updateSource (foundry.js:42953:19)
    at AmbientLight._draw (foundry.js:42873:10)
    at AmbientLight.draw (foundry.js:25199:16)
    at foundry.js:30825:26
    at EmbeddedCollection.map (commons.js:3332:26)
    at LightingLayer._draw (foundry.js:30823:32)
    at LightingLayer.draw (foundry.js:30349:16)
    at async InterfaceCanvasGroup.draw (foundry.js:26151:9)
    at async InterfaceCanvasGroup.draw (foundry.js:39270:5)
    at async Canvas.draw (foundry.js:23673:11)
    at async Scene.view (foundry.js:20901:7)
    at async Game.initializeCanvas (foundry.js:8095:20)
    at async Game.setupGame (foundry.js:7935:5)
    at async Game._initializeGameView (foundry.js:8999:5)
    at async Game.initialize (foundry.js:7825:5)
onError @ foundry.js:747
🎁call_wrapped @ libWrapper-wrapper.js:507
🎁Hooks.onError#lib-wrapper @ listeners.js:137
🎁Hooks.onError#0 @ libWrapper-wrapper.js:187
draw @ foundry.js:23676
await in draw (async)
view @ foundry.js:20901
initializeCanvas @ foundry.js:8095
await in initializeCanvas (async)
setupGame @ foundry.js:7929
await in setupGame (async)
_initializeGameView @ foundry.js:8999
_initializeView @ foundry.js:8975
initialize @ foundry.js:7825
await in initialize (async)
🎁call_wrapped @ libWrapper-wrapper.js:507
🎁libWrapperInit @ libWrapper-api.js:805
await in 🎁libWrapperInit (async)
🎁Game.prototype.initialize#0 @ libWrapper-wrapper.js:187
window.addEventListener.once @ foundry.js:79693
foundry.js:59187 Canvas drawing failed: Cannot read properties of undefined (reading 'heightWalls')
fetch @ foundry.js:59187
notify @ foundry.js:59114
error @ foundry.js:59150
onError @ foundry.js:748
🎁call_wrapped @ libWrapper-wrapper.js:507
🎁Hooks.onError#lib-wrapper @ listeners.js:137
🎁Hooks.onError#0 @ libWrapper-wrapper.js:187
draw @ foundry.js:23676
await in draw (async)
view @ foundry.js:20901
initializeCanvas @ foundry.js:8095
await in initializeCanvas (async)
setupGame @ foundry.js:7929
await in setupGame (async)
_initializeGameView @ foundry.js:8999
_initializeView @ foundry.js:8975
initialize @ foundry.js:7825
await in initialize (async)
🎁call_wrapped @ libWrapper-wrapper.js:507
🎁libWrapperInit @ libWrapper-api.js:805
await in 🎁libWrapperInit (async)
🎁Game.prototype.initialize#0 @ libWrapper-wrapper.js:187
window.addEventListener.once @ foundry.js:79693
foundry.js:747 TypeError: Cannot set properties of null (setting 'visible')
[No packages detected]
    at NotesLayer._deactivate (foundry.js:30889:26)
    at NotesLayer._deactivate (foundry.js:36177:11)
    at NotesLayer.deactivate (foundry.js:30474:10)
    at TokenLayer.activate (foundry.js:30438:76)
    at #initializeCanvasLayer (foundry.js:23846:11)
    at #initialize (foundry.js:23796:32)
    at Canvas.draw (foundry.js:23689:29)
    at async Scene.view (foundry.js:20901:7)
    at async Game.initializeCanvas (foundry.js:8095:20)
    at async Game.setupGame (foundry.js:7935:5)
    at async Game._initializeGameView (foundry.js:8999:5)
    at async Game.initialize (foundry.js:7825:5)
```

The following modules were loaded:
![image](https://user-images.githubusercontent.com/12570777/209060236-17f1ee87-7dfe-47be-84cd-8dfbdc83ad4a.png)
![image](https://user-images.githubusercontent.com/12570777/209060330-9aa6eee2-1184-4cd1-a6a8-67d8ba850e4b.png)

**Comment by Zamrod** [comment]
i also tested it with Wall Height off with the same issues

**Comment by caewok** [comment]
That is odd, as (obviously), I had it running fine in testing with walls.

**Comment by caewok** [comment]
Should be fixed in v0.3.1. Looks like Foundry tries to render existing sources before updating their line-of-sight. The result is that for existing scenes with at least light sources, the old data is used for the first render. Thus the error message about not finding "heightWalls," which is a new property. Thanks!
